packages/pages-shared/src/asset-server/rulesEngine.ts

```diff
@@ -48,9 +49,27 @@ export const generateRulesMatcher = <T>(
 				rule = rule.split(host_match[0]).join(`(?<${host_match[1]}>[^/.]+)`);
 			}
 
-			const path_matches = rule.matchAll(PLACEHOLDER_REGEX);
-			for (const path_match of path_matches) {
-				rule = rule.split(path_match[0]).join(`(?<${path_match[1]}>[^/]+)`);
+			let pathPart = split[0];
+			const queryParts = split.slice(1);
+
+			const path_matches = pathPart.matchAll(PLACEHOLDER_REGEX);
+			for (const pathMatch of path_matches) {
+				pathPart = pathPart
+					.split(pathMatch[0])
+					.join(`(?<${pathMatch[1]}>[^/?]+)`);
+			}
+			rule = pathPart;
```

Comment:
I think this overrides the work done on L49?

Reply:
[d35171d](https://github.com/cloudflare/wrangler2/pull/1863/commits/d35171d6e7d652284bbfc93706389d49ea282c8a)

Reply:
Actually.. that seems to have broken something, hold on

Reply:
Good catch, should be fixed!